scheduler-interface-spec.md

```diff
@@ -792,6 +792,31 @@ message ForgotAllocation {
 }
 ```
 
+#### Event Cache
```

Comment:
Can we rename

1.  `Event Cache` -> `Event Publisher` (cache is the core side thing, the plugin itself is to publish these events)
2. `EventMessage` -> `EventRecord` (message sounds more like a plain text, a record may have more rich format)

The Event Publisher is a SchedulerPlugin that exposes scheduling events from the scheduler core to the shim, which helps the end-user to retrieve these events from the shim side. An event is sent to the shim side through the callback in a form of an `EventRecord`.

Reply:
We should not reference any implementation unless we move the declaration of the callback into the SI. At this point I don't think we should do that.

Reply:
I keep the "Event Cache" title for this section, but `EventRecord` makes more sense to use both in the description and in the implementation.

Reply:
The event cache is not the plugin we do not expose the event cache as the plugin either. The cache is hidden inside the core. The plugin exposes a point to get events in the form of an `EventRecord`.
The comment must be something like the `Event Plugin`

Reply:
Ah, I understood now. Changed it `Event Plugin`.